sktime/forecasting/hf_transformers_forecaster.py

```diff
@@ -171,83 +178,79 @@ def __init__(
         self.peft_config = peft_config
 
     def _fit(self, y, X, fh):
-        from transformers import AutoConfig, Trainer, TrainingArguments
-
-        # Load model and extract config
-        config = AutoConfig.from_pretrained(self.model_path)
-
-        # Update config with user provided config
-        _config = config.to_dict()
-        _config.update(self._config)
-        _config["num_dynamic_real_features"] = X.shape[-1] if X is not None else 0
-        _config["num_static_real_features"] = 0
-        _config["num_dynamic_real_features"] = 0
-        _config["num_static_categorical_features"] = 0
-        _config["num_time_features"] = 0 if X is None else X.shape[-1]
-
-        if hasattr(config, "feature_size"):
-            del _config["feature_size"]
-
-        if fh is not None:
-            _config["prediction_length"] = max(
-                *(fh.to_relative(self._cutoff)._values + 1),
-                _config["prediction_length"],
-            )
+        from transformers import AutoConfig, PreTrainedModel, Trainer, TrainingArguments
 
-        config = config.from_dict(_config)
-        import transformers
+        if isinstance(self.model_path, PreTrainedModel):
+            self.model = self.model_path
+            self.config = self.model.config
```

Comment:
my bad, the reference error comes from here
```suggestion
            config = self.model.config
```